### PR TITLE
[libpq] Add support for dynamic builds with Visual Studio

### PR DESCRIPTION
**Pull request by SSE4**
* Fix source file lists to avoid link errors and to behave be as close as possible to the original MSVC projects.
* Generate more complete pg_config.h.
* Use .def file to make exported symbols work as expected.
* Add missing wldap32 library link in package_info().
* Enable dynamc build in conanfile.py

Specify library name and version:  **libpq/11.5**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

Original related bug report: https://github.com/bincrafters/conan-libpq/pull/25
Original PR: https://github.com/conan-io/conan-center-index/pull/143

### CODE REVIEW

File: recipes/libpq/all/conanfile.py [hunk 1]
```diff
@@ -13,7 +11,6 @@ class LibpqConan(ConanFile):
     homepage = "https://www.postgresql.org/docs/current/static/libpq.html"
     author = "Bincrafters <bincrafters@gmail.com>"
     license = "PostgreSQL"
-    exports_sources = ["CMakeLists.txt"]
```
Comment (Talkless) [inline recipes/libpq/all/conanfile.py]: CMakeLists.txt itself is not removed from repo..?

File: recipes/libpq/all/conanfile.py [hunk 2]
```diff
@@ -79,9 +77,25 @@ def _configure_cmake(self):
...
 
     def build(self):
-        if self.settings.os == "Windows" and self.settings.compiler == "Visual Studio":
-            cmake = self._configure_cmake()
-            cmake.build()
+        if self.settings.compiler == "Visual Studio":
+            # https://www.postgresql.org/docs/8.3/install-win32-libpq.html
+            # https://github.com/postgres/postgres/blob/master/src/tools/msvc/README
+            if not self.options.shared:
+                tools.replace_in_file(os.path.join(self._source_subfolder, "src", "tools", "msvc", "MKvcbuild.pm"),
```
Comment (Talkless) [inline recipes/libpq/all/conanfile.py]: What about zlib, openssl support? Doesn't it need any changes regards to these depenenceies? 

I see config_default.pl (https://github.com/postgres/postgres/blob/c114229ca2519620703a4be4e38181290cad8c0a/src/tools/msvc/config_default.pl#L23) has undefined openssl path by default.

File: recipes/libpq/all/conanfile.py (hunk 1 above)
Comment (SSE4) [reply recipes/libpq/all/conanfile.py]: I've kept it for the compatibility, but I can remove it, if needed

File: recipes/libpq/all/conanfile.py (hunk 2 above)
Comment (SSE4) [reply recipes/libpq/all/conanfile.py]: good point, I'll take a look to implement these options

File: recipes/libpq/all/conanfile.py (hunk 1 above)
Comment (SSE4) [reply recipes/libpq/all/conanfile.py]: removed all the CMake leftover

File: recipes/libpq/all/conanfile.py (hunk 2 above)
Comment (SSE4) [reply recipes/libpq/all/conanfile.py]: added openssl and zlib handling
also, there are other deps, but as they aren't handled for all platforms, we may introduce them eventually, on demand

**Pull request merged**